snippets/Timer.md

```diff
@@ -0,0 +1,49 @@
+### Timer
+
+Renders a timer component.
+
+- The timer state is initially set to zero 
+- When the `Tick!` button is clicked, `timer` is incremented periodically at the given `interval`
+- When the `Reset` button is clicked, the value of the timer is set to zero and the `setInterval` is cleared
+- The `setInterval` is cleared once the desired `time` is reached
+- `time` and `interval` are the required props
+
+```jsx
+class Timer extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {timer: 0}
+		this.interval = null
+	}
+
+	tick = () => {
+		this.reset()
+		this.interval = setInterval(() => {
+			if (this.state.timer < this.props.time) {
+				this.setState({timer: this.state.timer + 1})
```

Comment:
Since this is based off of the previous state it should use a state callback so that all updates will be consistent regardless of when they actually happen

```suggestion
				this.setState(({ timer }) => ({timer: timer + 1}))
```

Reply:
Didn't know about this. Thanks!